### PR TITLE
add: browserstack-service backwards compatibility

### PR DESCRIPTION
**Pull request by Ankit098**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
Making the browserstack-service run with older webdriverIO versions. 
Assigning `_config  = options` in constructor as in webdriverIO v5 the config is being passed in options, with config being undefined. 

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### Reviewers: @webdriverio/project-committers

<a href="https://gitpod.io/#https://github.com/webdriverio/webdriverio/pull/8374"></a>

### CODE REVIEW

**Review by christian-bromann — COMMENTED** [review]
Can we avoid the `| any` ?

Also we should enhance the global namespace that there could be a potential browser variable so we can avoid `this._browser = browser ? browser :`

File: packages/wdio-browserstack-service/src/launcher.ts [hunk 1]
```diff
@@ -20,10 +20,11 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
     browserstackLocal?: BrowserstackLocal
 
     constructor (
-        private _options: BrowserstackConfig,
+        private _options: BrowserstackConfig & Options.Testrunner,
         capabilities: Capabilities.RemoteCapability,
         private _config: Options.Testrunner
     ) {
+        this._config || (this._config = _options)
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/launcher.ts:27]: Can we add a comment that this is only for backward compatibility reasons?

File: packages/wdio-browserstack-service/src/service.ts [hunk 2]
```diff
@@ -57,7 +58,7 @@ export default class BrowserstackService implements Services.ServiceInstance {
     }
 
     before(caps: Capabilities.RemoteCapability, specs: string[], browser: Browser<'async'> | MultiRemoteBrowser<'async'>) {
-        this._browser = browser
+        this._browser = browser ? browser : (global as any).browser
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/service.ts:61]: same here

**Comment by Ankit098** [comment]
Hey, is this good for merge into main? Also could you please share an ETA for these changes to be included in the npm package..

**Comment by christian-bromann** [comment]
Thanks for the patch!

I plan to have a release next week.

**Comment by Ankit098** [comment]
Awesome! thanks.

**Pull request merged**